lib/msf/java/rmi/builder.rb

```diff
@@ -0,0 +1,96 @@
+# -*- coding: binary -*-
+
+require 'rex/java/serialization'
+
+module Msf
+  module Java
+    module Rmi
+      module Builder
+        # Builds a RMI header stream
+        #
+        # @param opts [Hash{Symbol => <String, Fixnum>}]
+        # @option opts [String] :signature
+        # @option opts [Fixnum] :version
+        # @option opts [Fixnum] :protocol
+        # @return [Rex::Proto::Rmi::Model::OutputHeader]
+        def build_header(opts = {})
```

Comment:
I was not familiar with keyword arguments. There is something which I like more from the Hash approach vs the keyword arguments.

Take for example this ```Msf::Java::Rmi::Builder#build_header``` method. It is used from the ```Msf::Java::Rmi::Client#send_header```. The ```send_header``` passes the ```opts``` to ```build_header``` so it can do its job, later he does its own job (which is send the header through network). With the current way ```send_header``` doesn't need ```connascence``` of ```build_header``` options (parameters):

```ruby
        def send_header(opts = {})
          nsock = opts[:sock] || sock
          stream = build_header(opts)
          nsock.put(stream.encode + "\x00\x00\x00\x00\x00\x00")
        end
```

With the ```opts``` way, the ```send_header``` methods) just gateways the ```opts``` hash. Is there a way to make the same thing with keyword arguments? Otherwise I guess I should also modify ```send_header``` to do:

```ruby
def send_header(nsock: sock, singature:, version:, protocol:)
    stream = build_header(
        signature: signature, 
         version: version, 
         protocol: protocol)
    nsock.put(stream.encode + "\x00\x00\x00\x00\x00\x00")
end
```

And the task of taking care of all the parameters could be even harder on complex domains like the ```Rex::Java::Serialization::Model``` one, where the usage of ```opts``` is kinda useful to build complex structures.